etc/systemd/system/linstor-satellite.service.d/override.conf

```
@@ -1,5 +1,5 @@
 [Service]
-Environment=LS_KEEP_RES=^xcp-persistent*
+Environment=LS_KEEP_RES=^xcp-persistent.*
```

The new regexp will still match `xcp-persistenttttttt`. Can't it be more precise?

---

Indeed, it could be:
```suggestion
Environment=LS_KEEP_RES=^xcp-persistent-.*
```

---

Is there actually an issue in being too broad? (Genuinely asking) If so, we could use a more precise regex like `xcp-persistent-([a-z0-9_]+-?)+` or `xcp-persistent(-[a-z0-9_]+)+` (both provides the same results). This would restrict matches to actually valid strings instead of `xcp-persistenttttttt` or `xcp-persistent-`, `xcp-persistent-database------`, etc.

---

To be clear: we can use `^xcp-persistent` as a regex here, in the LINSTOR use case it will work, this is equivalent to the suggestion with the `.*` that I recommend (it mainly avoids a change of regex if the parsing mode used changes on the controller side). I just want to avoid accepting "xcp-persisten", which was the case before. I just want to keep it simple and accept a volume as persistent as soon as it has the `xcp-persistent` prefix, I don't care what comes after that.

Regarding Mathieu's suggestion, I think it's good, but:
I had originally thought about the `$`, but I don't think it's useful. Basically, this commit just fixes a typo that has no real impact in production. A more specific commit message might contain: "Avoid detection of persistent volumes whose name contains only the prefix `xcp-persisten` instead of `xcp-persistent`". I prefer (for now) that we avoid being restrictive on the sequence of characters after this prefix.

---

That's why I approved in the first place. I'm ok with the broad regex, my suggestion was more to add a more precise proposition following Samuel's question. But the original proposition you made is ok for me.

---

And my comment was based on Ronan's initial message saying that matching `xcp-persistenttttttt` was not necessarily wanted :)

---

Yeah, the initial description is wrong. :stuck_out_tongue: